apidef/oas/middleware.go

```diff
@@ -1216,10 +1220,30 @@ type VirtualEndpoint struct {
 	RequireSession bool `bson:"requireSession,omitempty" json:"requireSession,omitempty"`
 }
 
+// MarshalJSON is a custom JSON marshaler for the VirtualEndpoint struct. It is implemented
+// to facilitate a smooth migration from deprecated fields that were previously used to represent
+// the same data.
+func (v *VirtualEndpoint) MarshalJSON() ([]byte, error) {
+	if v.FunctionName == "" && v.Name != "" {
+		v.FunctionName = v.Name
```

Comment:
This is probablly not great, marshaller writing out to `v` may cause side effects; all fields are primitives so it makes sense to make a copy into the Alias type?